examples/experiments/grounded_program_synthesis/README.md

```diff
@@ -0,0 +1,29 @@
+# Interpreter Grounded Program Synthesis
+*Program synthesis* is the task of automatically generating programs that solve a given task by satisfying an IO condition. In Neural Program Synthesis the synthesizer is a neural network which is a Language Model that takes in an input/output pair and tries to generate the program in the defined toy DSL's Grammar.
+
+## Toy List Manipulation DSL Grammar
+The DSL has the following grammar:
```

Comment:
In addition to the DSL grammar, add some snippets

Reply:
Added example snippets to showcase the atomic functions.